torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -1003,6 +1004,33 @@ Operation listAppend(const Node* node) {
   };
 }
 
+template <typename TList>
+Operation listExtend(const Node* node) {
+  return [](Stack& stack) {
+    TList a;
+    TList b;
+    pop(stack, a, b);
+
+    auto& vec_a = a->elements();
+    const auto& vec_b = b->elements();
+    vec_a.insert(vec_a.end(), vec_b.cbegin(), vec_b.cend());
+    return 0;
+  };
+}
+
+template <typename TList>
+Operation listCopy(const Node* node) {
+  return [](Stack& stack) {
+    TList list;
+    pop(stack, list);
+
+    const auto& vec = list->elements();
```

Comment:
Does actually perform a copy? 

```python
a = [1, 2, 3]
b = a.copy()
```

Won't creating an IValue from a reference to `a` result in it `b` sharing the same underlying data?

Reply:
It will indeed perform a *shallow* copy.

```
    const auto& vec = list->elements();
    auto out = vec;
```

`auto out` is deduced to be `std::vector<decl(vec)::value_type>`, which isn't a reference type, so `vec` gets copied via an assignment operator into `out`.  I believe this is an `IValue` c-tor that gets invoked in our case is `IValue::IValue(std::vector<int64_t> v)` (or a similar one corresponding to a given element type). We could explicitly pass a const reference (i.e. `const auto& vec`) into  the c-tor and we should **still** get a shallow copy. I do prefer explicit copies and let a compiler elide them (even lvalue ones), even though passing a const reference should (in theory) prevent us from going to the wrong c-tor even if `IValue::IValue(std::vector<int64_t> &v)` gets added.